forum/models/model_utils.py

```diff
@@ -776,3 +781,176 @@ def subscribe_user(
 def unsubscribe_user(user_id: str, source_id: str) -> None:
     """Unsubscribe a user from a source."""
     Subscriptions().delete_subscription(user_id, source_id)
+
+
+def get_user_voted_ids(user_id: str, vote: str) -> list[str]:
+    """Get the IDs of the posts voted by a user."""
+    if vote not in ["up", "down"]:
+        raise ValueError("Invalid vote type")
+
+    content_model = Contents()
+    contents = content_model.list()
+    voted_ids = []
+    for content in contents:
+        votes = content["votes"][vote]
+        if user_id in votes:
+            voted_ids.append(content["_id"])
+
+    return voted_ids
+
+
+def user_to_hash(
+    user: dict[str, Any], params: Optional[dict[str, Any]] = None
+) -> dict[str, Any]:
+    """
+    Converts user data to a hash
+    """
+    if params is None:
+        params = {}
+
+    hash_data = {}
+    hash_data["username"] = user["username"]
+    hash_data["external_id"] = user["external_id"]
+
+    comment_model = Comment()
+    thread_model = CommentThread()
+
+    if params.get("complete"):
+        subscribed_thread_ids = find_subscribed_threads(user["id"])
+        upvoted_ids = get_user_voted_ids(user["id"], "up")
+        downvoted_ids = get_user_voted_ids(user["id"], "down")
+        hash_data.update(
+            {
+                "subscribed_thread_ids": subscribed_thread_ids,
+                "subscribed_commentable_ids": [],
+                "subscribed_user_ids": [],
+                "follower_ids": [],
+                "id": user["id"],
+                "upvoted_ids": upvoted_ids,
+                "downvoted_ids": downvoted_ids,
+                "default_sort_key": user["default_sort_key"],
+            }
+        )
+
+    if params.get("course_id"):
+        if params.get("group_ids"):
+            specified_groups_or_global = params["group_ids"] + [None]
+            threads = thread_model.list(
+                author_id=user["id"],
+                course_id=params["course_id"],
+                anonymous=False,
+                anonymouse_to_peers=False,
+            )
+            thread_ids = [str(thread["_id"]) for thread in threads]
+            group_query = {
+                "_id": {"$in": [ObjectId(tid) for tid in thread_ids]},
```

Comment:
we can use list of thread Object ids directly in query, instead of make a str list and then objectId list from str dict.
```
group_query = {
        "_id": {"$in": [thread["_id"] for thread in threads]},
```